snapcraft/internal/common.py

```diff
@@ -255,3 +255,39 @@ def format_output_in_columns(elements_list, max_width=MAX_CHARACTERS_WRAP,
         result_output.append(sep.join(candidate_output[i]))
 
     return result_output
+
+
+def get_include_paths(root, arch_triplet):
+    paths = [
+        os.path.join(root, 'include'),
+        os.path.join(root, 'usr', 'include'),
+        os.path.join(root, 'include', arch_triplet),
+        os.path.join(root, 'usr', 'include', arch_triplet),
+    ]
+
+    return [p for p in paths if os.path.exists(p)]
+
+
+def get_library_paths(root, arch_triplet):
+    paths = [
+        os.path.join(root, 'lib'),
+        os.path.join(root, 'usr', 'lib'),
+        os.path.join(root, 'lib', arch_triplet),
+        os.path.join(root, 'usr', 'lib', arch_triplet),
+    ]
+
+    return [p for p in paths if os.path.exists(p)]
+
+
+def combine_paths(paths, prepend, separator):
+    paths = ['{}{}'.format(prepend, p) for p in paths]
+    return separator.join(paths)
+
+
+def format_path_variable(envvar, paths, prepend, separator):
+    if not paths:
+        return ''
```

Comment:
El 24/06/16 a las 14:49, Kyle Fazzari escribió:
> In snapcraft/internal/common.py
> <https://github.com/ubuntu-core/snapcraft/pull/566#discussion_r68434984>:
> 
>> +        os.path.join(root, 'usr', 'lib'),
>> +        os.path.join(root, 'lib', arch_triplet),
>> +        os.path.join(root, 'usr', 'lib', arch_triplet),
>> +    ]
>> +
>> +    return [p for p in paths if os.path.exists(p)]
>> +
>> +
>> +def combine_paths(paths, prepend, separator):
>> +    paths = ['{}{}'.format(prepend, p) for p in paths]
>> +    return separator.join(paths)
>> +
>> +
>> +def format_path_variable(envvar, paths, prepend, separator):
>> +    if not paths:
>> +        return ''
> 
> Ah, yeah good catch. Honestly though I don't think we need to check it
> at all here-- if the caller asked to format an environment variable out
> of the paths, they should probably know that the paths are good,
> otherwise there's nothing reasonable to return. If you want ValueError
> raised though, that's fine too.

There's nothing wrong with raising a nice exception so the user of the
API can fix fast ;-)